### PR TITLE
Fix __SOFTFP__ case for arm getcontext

### PR DESCRIPTION
**Pull request by am11**
See https://github.com/libunwind/libunwind/pull/382#issuecomment-1204408556 for context.

### CODE REVIEW

**Comment by am11** [comment]
cc @vtjnash, @janvorli

File: src/dwarf/Gfind_proc_info-lsb.c [hunk 1]
```diff
@@ -135,9 +135,9 @@ load_debug_frame (const char *file, char **buf, size_t *bufsize, int is_local,
 #if defined(SHF_COMPRESSED)
   if (shdr->sh_flags & SHF_COMPRESSED)
     {
-      unsigned long destSize;
       Elf_W (Chdr) *chdr = (shdr->sh_offset + ei.image);
 #ifdef HAVE_ZLIB
+      unsigned long destSize;
```
Comment (am11) [inline src/dwarf/Gfind_proc_info-lsb.c:140]: Unrelated change: gcc was warning that destSize is unused (it is unused outside of HAVE_ZLIB branch).

**Comment by am11** [comment]
FWIW,  I compared macros using:
`echo | arm-linux-gnueabi-gcc -dM -E - | grep -iE 'arm|fpu|thumb|neon|soft'`
vs.
`echo | arm-linux-gnueabihf-gcc -dM -E - | grep -iE 'arm|fpu|thumb|neon|soft'`

**Review by vtjnash — COMMENTED** [review]
SGTM
